### PR TITLE
Drop dependency on deprecated gulp-util

### LINKED ISSUE #9: Replace deprecated dependency gulp-util
Replace deprecated dependency gulp-util

[`gulp-util`](https://www.npmjs.com/package/gulp-util) has been deprecated recently. Continuing to use this dependency may prevent the use of your library with the latest release of Gulp 4 so **it is important to replace `gulp-util`**.

The [README.md](https://github.com/gulpjs/gulp-util) lists alternatives for all the components so a simple replacement should be enough.

Your package is popular but still relying on `gulp-util`, it would be good to publish a fixed version to npm as soon as possible.

See:
- https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
- https://github.com/gulpjs/gulp-util/issues/143

### PR DESCRIPTION
**Pull request by Teamop**
closes #9

### CODE REVIEW

**Comment by Teamop** [comment]
The Ci passed on Node 8, but it failed on Node 6 and Node 4, after I researched, this issue is related to typescript, right now, the code can just passed on TS 2.4.2, will break on the next versions, so I will just pin the TS to 2.4.2.

Issues got from the Travis CI, which related to params.spec.ts
<details>
  <summary>Issues(partial)</summary>
TSError: ⨯ Unable to compile TypeScript
spec/params.spec.ts (19,36): Value of type '{ <T>(value: T): T; (): undefined; }' has no properties in common with type '{ encoding?: null | undefined; thisArg?: any; }'. Did you mean to call it? (2560)
    at getOutput (/home/travis/build/mcmath/gulp-transform/node_modules/ts-node/src/index.ts:307:15)
    at /home/travis/build/mcmath/gulp-transform/node_modules/ts-node/src/index.ts:336:16
</details>

**Comment by dex4er** [comment]
@mcmath ping?

File: package.json [hunk 1]
```diff
@@ -35,18 +35,20 @@
...
+    "tslib": "^1.7.1",
+    "vinyl": "^2.1.0"
   },
   "devDependencies": {
     "@types/chai": "^4.0.1",
     "@types/chai-as-promised": "0.0.31",
     "@types/lodash": "^4.14.71",
     "@types/mocha": "^2.2.41",
+    "@types/node": "*",
+    "@types/plugin-error": "^0.1.0",
```
Comment (demurgos) [inline package.json]: `plugin-error` bundles its own types: no need for the old `@types/plugin-error` package.

**Review by demurgos — APPROVED** [review]
Looks good to me. Just remove `@types/plugin-error`.

@mcmath 
Could we have this merged and published?

File: package.json (hunk 1 above)
Comment (Teamop) [reply package.json]: thanks @demurgos, removed

**Comment by eduardoconceicao** [comment]
@demurgos Can we get this merged?

**Comment by demurgos** [comment]
I would like to, but I am not a maintainer of this repo.

/cc @mcmath

**Pull request closed without merging**